Studies/python/InputsProducer.py

```diff
@@ -46,45 +45,40 @@ def DefineVariables(sample_name, parity, use_deepTau_ordering) :
         df = df.Filter('evt % 2 == {}'.format(parity))
 
     df =  df.Define('deepFlavour_bVSall', 'MakeDeepFlavour_bVSall(jets_deepFlavour_b, jets_deepFlavour_bb, jets_deepFlavour_lepb)')
-    # .Define('n_jets', 'jets_deepFlavourOrderedIndex.size()')
-
 
     if use_deepTau_ordering :
-        df = df.Define('tau_indeces_deepTau', 'getSignalTauIndicesDeep_Tau(lep_p4, byDeepTau2017v2p1VSjetraw)') \
-                .Define('jets_deepFlavourOrderedIndex', 'CreateOrderedIndex(jets_p4, deepFlavour_bVSall, true, tau_indeces_deepTau, lep_p4, {})'.format(max_jet)) \
-                .Define('n_jets', 'jets_deepFlavourOrderedIndex.size()') \
-                .Define('htt_p4', 'getHTTp4(lep_p4, tau_indeces_deepTau)') \
-                .Define('htt_scalar_pt', 'getHTTScalarPt(lep_p4, tau_indeces_deepTau)') \
-                .Define('htt_pt', 'htt_p4.pt()') \
-                .Define('htt_eta', 'htt_p4.eta()')
-
-    elif use_deepTau_ordering == False:
+        df = df.Define('tau_indices_sel', 'getSignalTauIndicesDeep_Tau(lep_p4, byDeepTau2017v2p1VSjetraw, byDeepTau2017v2p1VSeraw, byDeepTau2017v2p1VSmuraw)') \
+               .Filter('tau_indices_sel.size() == 2') \
+               .Define('jets_deepFlavourOrderedIndex', 'CreateOrderedIndex(jets_p4, deepFlavour_bVSall, true, tau_indices_sel, lep_p4, {})'.format(max_jet)) \
+               .Define('htt_scalar_pt', 'getHTTScalarPt(lep_p4, tau_indices_sel)') \
+               .Define('htt_p4', 'getHTTp4(lep_p4, tau_indices_sel)')
+
+    else:
         df = df.Define('tau_indeces_gen', 'getSignalTauIndices_Gen(lep_p4, lep_genTauIndex)') \
```

Comment:
tau_indeces_gen -> tau_indeces_sel. Definition of jets_deepFlavourOrderedIndex, htt_scalar_pt and htt_p4 shouldn't directly depend on tau ordering.